tests/rptest/tests/cloud_retention_test.py

```diff
@@ -8,60 +8,84 @@
+
+    def setUp(self):
+        # Defer cluster startup so we can tweak configs based on the type of
+        # test run.
+        pass
+
+    @cluster(num_nodes=4)
+    @matrix(max_consume_rate_mb=[20, None])
+    @skip_debug_mode
+    def test_cloud_retention(self, max_consume_rate_mb):
+        """
+        Test cloud retention with an ongoing workload.
+        """
+        if self.redpanda.dedicated_nodes:
+            num_partitions = 10
+            msg_size = 128 * 1024
+            msg_count = int(100 * 1024 * 1024 * 1024 / msg_size)
+            segment_size = 10 * 1024 * 1024
+        else:
+            num_partitions = 1
+            msg_size = 1024
+            msg_count = int(1024 * 1024 * 1024 / msg_size)
+            segment_size = 1024 * 1024
+
+        si_settings = SISettings(log_segment_size=segment_size)
+        self.redpanda.set_si_settings(si_settings)
+
         extra_rp_conf = dict(retention_local_target_bytes_default=self.
-                             default_retention_segments * self.segment_size,
+                             default_retention_segments * segment_size,
                              log_segment_size_jitter_percent=5,
                              group_initial_rebalance_delay=300,
                              cloud_storage_segment_max_upload_interval_sec=60,
                              cloud_storage_housekeeping_interval_ms=10_000)
+        self.redpanda.add_extra_rp_conf(extra_rp_conf)
 
-        super(CloudRetentionTest, self).__init__(
-            test_context=test_context,
-            node_prealloc_count=1,
-            num_brokers=3,
-            si_settings=SISettings(log_segment_size=self.segment_size),
-            extra_rp_conf=extra_rp_conf)
+        self.redpanda.start()
+        self._create_initial_topics()
 
-    @cluster(num_nodes=4)
-    def test_cloud_retention(self):
-        """
-        Test cloud retention with an ongoing workload.
-        """
-        num_partitions = 1
         rpk = RpkTool(self.redpanda)
         rpk.create_topic(topic=self.topic_name,
                          partitions=num_partitions,
                          replicas=3,
                          config={
                              "cleanup.policy": TopicSpec.CLEANUP_DELETE,
-                             "retention.bytes": 5 * self.segment_size,
+                             "retention.bytes": 5 * segment_size,
                          })
 
-        msg_size = 1024
-        msg_count = int(1024 * 1024 * 1024 / msg_size)
         producer = KgoVerifierProducer(self.test_context,
                                        self.redpanda,
                                        self.topic_name,
                                        msg_size=msg_size,
                                        msg_count=msg_count,
                                        custom_node=self.preallocated_nodes)
 
-        max_read_msgs = 25000
+        if max_consume_rate_mb is None:
```

Comment:
nit: prefer to pull these kinds of conditionals up to the top of the function with the rest of the parameters

Reply:
Done